packages/tx/src/eip4844Transaction.ts

```diff
@@ -172,6 +176,28 @@ export class BlobEIP4844Transaction extends BaseTransaction<BlobEIP4844Transacti
   }
 
   public static fromTxData(txData: BlobEIP4844TxData, opts?: TxOptions) {
+    if (txData.blobsData !== undefined) {
+      if (txData.blobs !== undefined) {
+        throw new Error('cannot have both raw blobs data and encoded blobs in constructor')
+      }
+      if (txData.kzgCommitments !== undefined) {
+        throw new Error('cannot have both raw blobs data and KZG commitments in constructor')
+      }
+      if (txData.versionedHashes !== undefined) {
+        throw new Error('cannot have both raw blobs data and versioned hashes in constructor')
+      }
+      if (txData.kzgProofs !== undefined) {
+        throw new Error('cannot have both raw blobs data and KZG proofs in constructor')
+      }
+      txData.blobs = getBlobs(txData.blobsData.reduce((acc, cur) => acc + cur))
```

Comment:
nice, so blobsData can be any length array  that needs to be chunked into blobs!